debug_module.tex

```diff
@@ -229,13 +217,32 @@ \section{Abstract Commands} \label{abstractcommands}
     \end{tabulary}
 \end{table}
 
+\begin{commentary}
+    The Abstract Command interface is designed to allow a debugger to write
+    commands as fast as possible, and then later check whether they completed
+    without error.  In the common case the debugger will be much slower than
+    the target and commands succeed, which allows for maximum throughput. If
+    there is a failure, the interface ensures that no commands execute after
+    the failing one.  To discover which command failed, the debugger has to
+    look at the state of the DM (e.g.\ contents of \Rdatazero) or hart (e.g.\ 
+    contents of a register modified by a Program Buffer program) to determine
+    which one failed.
+\end{commentary}
+
 Before starting an abstract command, a debugger must ensure that \Fhaltreq,
 \Fresumereq, and \Fackhavereset are all 0.
 
 While an abstract command is executing (\Fbusy in \Rabstractcs is high), a
 debugger must not change \Fhartsel, and must not write 1 to \Fhaltreq,
 \Fresumereq, \Fackhavereset, \Fsetresethaltreq, or \Fclrresethaltreq.
 
+To abort an abstract command, the debugger first resets the hart (using
+\Fhartreset or \Fndmreset), and then the Debug Module (using \Fdmactive).
```

Comment:
We may also want to say this is not expected scenario and should only happen due to something wrong (not normal SW debugging scenarios). It may not actually resolve the situation, depending on what went wrong (e.g. locked up bus is not resolved by this procedure).

Reply:
Maybe we should say "If an abstract command does not complete in the expected time and appears to be hung, the following procedure can be attempted to abort the command:..."

Reply:
Done.